ddtrace/_logger.py

```diff
@@ -62,11 +66,21 @@ def _configure_ddtrace_file_logger(logger):
         ddtrace_file_handler = RotatingFileHandler(
             filename=log_path, mode="a", maxBytes=max_file_bytes, backupCount=num_backup
         )
-
         log_format = "%(asctime)s %(levelname)s [%(name)s] [%(filename)s:%(lineno)d] - %(message)s"
         log_formatter = logging.Formatter(log_format)
-
         ddtrace_file_handler.setLevel(file_log_level_value)
         ddtrace_file_handler.setFormatter(log_formatter)
         logger.addHandler(ddtrace_file_handler)
         logger.debug("ddtrace logs will be routed to %s", log_path)
+
+
+def _configure_log_injection():
+    """
+    Ensures that logging is patched before we inject trace information into logs.
+    """
+    from ddtrace import patch
+
+    patch(logging=True)
+    ddtrace_logger = logging.getLogger("ddtrace")
+    if ddtrace_logger.hasHandlers():
+        ddtrace_logger.handlers[0].setFormatter(logging.Formatter(DD_LOG_FORMAT))
```

Comment:
I'm pretty confident user code will always run after `_configure_log_injection()` is called since it's called on tracer module load and sitecustomize.py.